components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -197,12 +197,7 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
                 regexp = registeredCallbackUrl.substring(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX.length());
             }
 
-            if (regexp != null && callbackURI.matches(regexp)) {
-                validationResponseDTO.setValidClient(true);
-                validationResponseDTO.setApplicationName(appDO.getApplicationName());
-                validationResponseDTO.setCallbackURL(callbackURI);
-                return validationResponseDTO;
-            } else if (appDO.getCallbackUrl().equals(callbackURI)) {
+            if (validateCallbackURI(regexp, callbackURI, appDO)) {
```

Comment:
Since regex is derived from app config, shall we that logic also into the private method please?

Reply:
So basically validateCallbackURI logic only takes the callbackURI from the request and appDO object

Reply:
done